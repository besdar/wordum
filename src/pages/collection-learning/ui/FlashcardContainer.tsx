import React from 'react';
import {Grid} from '../../../shared/ui/Grid';
import {ViewStyle} from 'react-native';

type Props = {
  children: React.ReactNode;
  borderColor?: ViewStyle['borderColor'];
};

export const FlashcardContainer = ({children, borderColor}: Props) => (
  <Grid
    direction="column"
    rowGap={10}
    fillAwailableSpace
    alignItems="center"
    justifyContent="center"
    surfaceProps={{style: {borderRadius: 10, borderColor}}}>
    {children}
  </Grid>
);
