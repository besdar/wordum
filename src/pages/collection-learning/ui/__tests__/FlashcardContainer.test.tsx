import {render} from '@testing-library/react-native';
import {FlashcardContainer} from '../FlashcardContainer';
import {View} from 'react-native';
import React from 'react';

describe('FlashcardContainer', () => {
  it('should match the snapshot', () => {
    const {toJSON} = render(
      <FlashcardContainer>
        <View />
      </FlashcardContainer>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
