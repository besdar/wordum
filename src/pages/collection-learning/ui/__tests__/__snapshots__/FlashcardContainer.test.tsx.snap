// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardContainer should match the snapshot 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(247, 243, 249)",
      "borderRadius": 10,
      "flexGrow": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
  testID="surface-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(247, 243, 249)",
        "borderColor": undefined,
        "borderRadius": 10,
        "columnGap": undefined,
        "display": "flex",
        "flex": undefined,
        "flexBasis": undefined,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "gap": undefined,
        "justifyContent": "center",
        "padding": undefined,
        "rowGap": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="surface"
  >
    <View />
  </View>
</View>
`;
